p/scripts/main.js

```diff
@@ -697,19 +697,54 @@ function user_filter(key) {
 	}
 }
 
+function show_share_menu(el) {
+	const itemId = el.closest('.flux').id;
+	const templateId = 'share_article_template';
+	const id = itemId;
+	const flux_header_el = el.closest('.flux');
+	const title_el = flux_header_el.querySelector('.item.titleAuthorSummaryDate .item-element.title');
+	const websiteName = ' - ' + flux_header_el.querySelector('.flux_header').dataset.websiteName;
+	const articleAuthors = flux_header_el.querySelector('.flux_header').dataset.articleAuthors;
+	let articleAuthorsText = '';
+	if (articleAuthors.trim().length > 0) {
+		articleAuthorsText = ' (' + articleAuthors + ')';
+	}
+	const link = title_el.href;
+	const title = title_el.textContent;
+	const titleText = title;
+	const div = el.parentElement;
+	const dropdownMenu = div.querySelector('.dropdown-menu');
+	const template = document.getElementById(templateId).innerHTML
+		.replace(/--entryId--/g, id)
+		.replace(/--link--/g, link)
+		.replace(/--titleText--/g, titleText)
+		.replace(/--websiteName--/g, websiteName)
+		.replace(/--articleAuthors--/g, articleAuthorsText);
+	if (!dropdownMenu) {
```

Comment:
Could this test be located at the top of the function, to avoid doing a lot of wasted work?

Reply:
I was imagining an early `return` statement to avoid a big `if`, but it is also fine like this

Reply:
Done.

Then a refactoring is needed here too:
https://github.com/FreshRSS/FreshRSS/blob/24d10feebd0c9802b1082e88bf165902885591aa/p/scripts/main.js#L914-L936

Should I do it with this PR or a new PR?

Reply:
ok. You merged it already :D